### PR TITLE
Upgrade to elasticsearch 7

### PR DESCRIPTION
**Pull request by ishitatsuyuki**
Additional changes required for upgrade (already executed for tsg-ut server):
- Set [sysctl](https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html)
- Wipe elasticsearch volume (to create the index from scratch)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -46,6 +45,8 @@ services:
     command: --profile=1 --slowms=100 --bind_ip_all --replSet rs0
     volumes:
       - mongodata:/data/db
+    ports:
```
Comment (ishitatsuyuki) [inline docker-compose.yml:48]: I don't know when this change has been introduced

File: connector/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
-FROM alpine:3.9
+FROM rwynn/monstache:6.7.1
 
-RUN apk add --update python3 py3-pip mongodb bash curl && \
-	python3 -m pip install 'mongo-connector[elastic5]' 'elastic2-doc-manager[elastic5]' && \
+RUN apk add --no-cache bash curl && \
```
Comment (ishitatsuyuki) [inline connector/Dockerfile:3]: `--no-cache` is best practice for Docker

File: docker-compose.yml [hunk 3]
```diff
@@ -36,7 +34,8 @@ services:
   elasticsearch:
     build: elasticsearch
     environment:
-      - ES_JAVA_OPTS=-Xms500m -Xmx500m
+      - ES_JAVA_OPTS=-Xms640m -Xmx640m
```
Comment (ishitatsuyuki) [inline docker-compose.yml:37]: 640M seemed to give a good balance of GC overhead and swapping (still swaps a lot when reindexing though)

File: viewer/viewer.rb [hunk 4]
```diff
@@ -73,7 +73,7 @@ def search(params)
     lte: params[:max_ts],
   }
 
-  uri = URI.parse('http://elasticsearch:9200/slack_logger/messages/_search')
+  uri = URI.parse('http://elasticsearch:9200/slack_logger.messages/_search')
```
Comment (ishitatsuyuki) [inline viewer/viewer.rb:76]: Index name changed due to the switch to monstache

File: connector/run.sh [hunk 5]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
 mongo mongo:27017 <<< "rs.initiate({_id: 'rs0', members: [{_id: 0, host: 'mongo:27017'}], protocolVersion: 1}, {force: true})"
-curl -XPUT 'http://elasticsearch:9200/slack_logger' -d '{
+# FIXME: We should healthcheck elasticsearch or the commands here might fail!
+curl -H "Content-Type: application/json" -XPUT 'http://elasticsearch:9200/slack_logger.messages' -d '{
```
Comment (ishitatsuyuki) [inline connector/run.sh:5]: ElasticSearch 6.0+ requires content type for API calls

File: connector/monstache.toml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+elasticsearch-max-conns = 2
+elasticsearch-retry = true
+direct-read-split-max = 2
```
Comment (ishitatsuyuki) [inline connector/monstache.toml:3]: This option strongly affects memory usage, default is 9 but 2 is probably a good tradeoff between memory and reindexing performance

File: connector/Dockerfile (hunk 2 above)
Comment (cookie-s) [reply connector/Dockerfile:3]: そうなんや

**Review by cookie-s — APPROVED** [review]
はいプロ

[MEMO]
https://slack-log.tsg.ne.jp/C0XFS4J0Z/1605003924.099100
https://slack-log.tsg.ne.jp/C0XFS4J0Z/1605004176.099300

んごこねくたーがもう更新されてないっぽい
https://github.com/yougov/mongo-connector

File: docker-compose.yml (hunk 1 above)
Comment (cookie-s) [reply docker-compose.yml:48]: ありがとう、なんかずっとみんな「なんやこいつ」っつってたんですよね

File: connector/monstache.toml (hunk 6 above)
Comment (cookie-s) [reply connector/monstache.toml:3]: はいプロ

**Review by cookie-s — APPROVED** [review]

**Comment by cookie-s** [comment]
マージしちゃお

**Pull request merged**